### PR TITLE
Update Authenticator Test Suite for Rails 5 Upgrade

### PR DESCRIPTION
**Pull request by micahlee**
*NOTE:* This PR merges into the `rails-5-again` branch, not `master`.

This PR updates the authenticator test suite to make the necessary fixes for the Rails 5 upgrade.

### CODE REVIEW

**Comment by micahlee** [comment]
Yeah, that seems to also be the case for the `rails-5-again` branch as well: https://jenkins.conjur.net/job/cyberark--conjur/job/rails-5-again/

The particular error is because the GKE coverage results are missing.

File: Jenkinsfile [hunk 1]
```diff
@@ -49,9 +49,9 @@ pipeline {
         stage('RSpec') {
           steps { sh 'ci/test rspec' }
         }
-        // stage('Authenticators') {
-        //   steps { sh 'ci/test cucumber_authenticators' }
-        // }
+        stage('Authenticators') {
+          steps { sh 'ci/test cucumber_authenticators' }
+        }
```
Comment (orenbm) [inline Jenkinsfile]: @micahlee please note that this has changed so you'll have some conflicts. See latest jenkinsfile -https://github.com/cyberark/conjur/blob/master/Jenkinsfile#L60

File: cucumber/authenticators/features/authn_status.feature [hunk 2]
```diff
@@ -107,13 +107,21 @@ Feature: Authenticator status check
     And the authenticator status check fails with error "UserNotAuthorizedInConjur: CONJ00006E"
 
   Scenario: An authenticator without an implemented status check returns 503
-    Given I login as "alice"
+    Given a policy:
+    """
+    - !user alice
+    """
+    And I login as "alice"
```
Comment (orenbm) [inline cucumber/authenticators/features/authn_status.feature]: @micahlee can you explain why this is part of the PR? is this change related to the Rails 5 upgrade?

File: Jenkinsfile (hunk 1 above)
Comment (micahlee) [reply Jenkinsfile]: Yeah, we are aware that were going to have some heavy rebasing to do. A lot has been going on in the testing framework lately! 🙂

File: cucumber/authenticators/features/authn_status.feature (hunk 2 above)
Comment (micahlee) [reply cucumber/authenticators/features/authn_status.feature]: Yes. When we started debugging test failures we discovered that we had quite a few scenarios that assumed a deterministic order, and that there was database state from prior runs.

We have started correcting this by doing more to ensure that each scenario "starts from scratch" and defines the entire scenario context itself.

These two scenarios are examples that assumed the user `alice` was already created in the database. So they now just need to specify that explicitly in the scenario by defining the user in policy.

Comment (orenbm) [reply cucumber/authenticators/features/authn_status.feature]: Nice, thanks!

When you create a PR to master please write this in a commit so it's clear what the change is . thanks for this!
